api/config/v1alpha1/envoygateway_types.go

```diff
@@ -139,11 +139,28 @@ const (
 	RedisBackendType RateLimitDatabaseBackendType = "Redis"
 )
 
-// RateLimitRedisSettings defines the configuration for connecting to
-// a Redis database.
+// RedisTLSSettings defines the TLS configuration for connecting to redis database.
+type RedisTLSSettings struct {
+	// Auth defines auth configuration to enable password-only authentication to the redis host.
+	//
+	// +optional
+	Auth string `json:"auth,omitempty"`
+
+	// CertificateRef defines the client certificate reference kubernetes secret for TLS connections.
+	//
+	// +optional
```

Comment:
optional fields are always pointer based

Reply:
from k8s.io
```golang
// A single application container that you want to run within a pod.
type Container struct {
	// Name of the container specified as a DNS_LABEL.
	// Each container in a pod must have a unique name (DNS_LABEL).
	// Cannot be updated.
	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
	// Container image name.
	// More info: https://kubernetes.io/docs/concepts/containers/images
	// This field is optional to allow higher level config management to default or override
	// container images in workload controllers like Deployments and StatefulSets.
	// +optional
	Image string `json:"image,omitempty" protobuf:"bytes,2,opt,name=image"`
```
Is pointer unnecessary?

Reply:
this might be because `name=image`

Reply:
using pointer